updateSettings new streaming parameters : trackSwitchMode and selectionModeForInitialTrack

jeffcunat opened the pull request:
This PR adds two new settings parameters (may have been forgotten when new **updateSettings** function was created in v3)
```
{ streaming: 
  {
    trackSwitchMode: {audio: audioMode , video: videoMode},
    selectionModeForInitialTrack: mode,
  }
}
```
and deprecates setTrackSwitchModeFor and setSelectionModeForInitialTrack functions (still usable though)

dsilhavy commented:
@jeffcunat Thanks for this PR. Minor issues:

- BufferController and ScheduleController still have a reference to `MediaController.TRACK_SWITCH_MODE_ALWAYS_REPLACE` instead of `Constants.TRACK_SWITCH_MODE_ALWAYS_REPLACE`
- Same for `unit/streaming.controllers.MediaController.js`

jeffcunat commented:
@dsilhavy I have fixed the minor issues you mentioned.
What is also different with this PR is that only audio and video type can be used for the new settings streaming.trackSwitchMode (before any type could be used). It seems that it is not relevant for text. But could you confirm ?

Unit tests were done with a fake 'test' type so now I have also added specific audio and video unit tests that test default settings values and their update.

dsilhavy commented:
@jeffcunat Thanks

Regarding your question: BufferController also checks the trackSwitchMode for fragmentedText. However, ranges.length is 0 for the streams I tested so it does not really affect anything. I think it is fine to limit this to audio and video.
